lib/SIL/SILDifferentiabilityWitness.cpp

```diff
@@ -0,0 +1,42 @@
+//===--- SILDifferentiabilityWitness.cpp - Differentiability witnesses ----===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2019 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+#define DEBUG_TYPE "sil-differentiability-witness"
+
+#include "swift/SIL/SILDifferentiabilityWitness.h"
+#include "swift/SIL/SILModule.h"
+
+using namespace swift;
+
+SILDifferentiabilityWitness *SILDifferentiabilityWitness::create(
+    SILModule &module, SILLinkage linkage, SILFunction *originalFunction,
+    IndexSubset *parameterIndices, IndexSubset *resultIndices,
+    GenericSignature *derivativeGenSig, SILFunction *jvp, SILFunction *vjp,
+    bool isSerialized, DeclAttribute *attribute) {
+  void *buf = module.allocate(sizeof(SILDifferentiabilityWitness),
+                              alignof(SILDifferentiabilityWitness));
+  auto *diffWitness = ::new (buf) SILDifferentiabilityWitness(
```

Comment:
```suggestion
  auto *diffWitness = new (module) SILDifferentiabilityWitness(
```

Reply:
Done in https://github.com/apple/swift/pull/27487/commits/2073f863826fb305c1fc105a520d84de1e1463ec.